### PR TITLE
feat(modals): add aria-attributes

### LINKED ISSUE #4114: feat(modals): add aria-attributes
Need add aria-attributes to HTML code in `modals` component.

- `role="dialog"` identifies the element that serves as the dialog container
- `aria-modal` tells assistive technologies that the windows underneath the current dialog are not available for interaction
- `aria-labelledby` identifies the element (or elements) that labels the current element

Took as a basis - https://www.w3.org/TR/2017/NOTE-wai-aria-practices-1.1-20171214/examples/dialog-modal/dialog.html

### PR DESCRIPTION
**Pull request by tarusin**
Closes #4114

# PR Checklist
Before creating new PR, please take a look at checklist below to make sure that you've done everything that needs to be done before we can merge it.

 - [x] read and followed the [CONTRIBUTING.md](https://github.com/valor-software/ngx-bootstrap/blob/development/CONTRIBUTING.md) guide.
 - [x] built and tested the changes locally.
 - [x] added/updated API documentation.
 - [x] added/updated demos.

### CODE REVIEW

File: src/modal/modal-container.component.ts [hunk 1]
```diff
@@ -18,7 +18,7 @@ import { isBs3 } from '../utils/theme-provider';
 @Component({
   selector: 'modal-container',
   template: `
-    <div [class]="'modal-dialog' + (config.class ? ' ' + config.class : '')" role="document">
+    <div [class]="'modal-dialog' + (config.class ? ' ' + config.class : '')">
```
Comment (IlyaSurmay) [inline src/modal/modal-container.component.ts]: why did you remove `role="document"?` original bootstrap' modal has it https://getbootstrap.com/docs/4.0/components/modal/#modal-components

File: demo/src/app/components/+modal/demos/auto-shown/auto-shown.html [hunk 2]
```diff
@@ -1,9 +1,10 @@
 <button type="button" class="btn btn-primary" (click)="showModal()">Render auto-shown modal</button>
-<div *ngIf="isModalShown" [config]="{ show: true }" (onHidden)="onHidden()" bsModal #autoShownModal="bs-modal" class="modal fade" tabindex="-1" role="dialog" aria-hidden="true">
+<div *ngIf="isModalShown" [config]="{ show: true }" (onHidden)="onHidden()" bsModal #autoShownModal="bs-modal" class="modal fade"
+     tabindex="-1" role="dialog" aria-hidden="true" aria-labelledby="dialog-auto-name" aria-modal="true">
```
Comment (IlyaSurmay) [inline demo/src/app/components/+modal/demos/auto-shown/auto-shown.html]: aria-hidden attribute has to be removed or changed to `false` when the modal is opened

Comment (IlyaSurmay) [reply demo/src/app/components/+modal/demos/auto-shown/auto-shown.html]: this is for all the demos you've changed in this PR

**Review by IlyaSurmay — CHANGES_REQUESTED** [review]

File: demo/src/app/components/+modal/demos/auto-shown/auto-shown.html [hunk 3]
```diff
@@ -1,9 +1,10 @@
 <button type="button" class="btn btn-primary" (click)="showModal()">Render auto-shown modal</button>
-<div *ngIf="isModalShown" [config]="{ show: true }" (onHidden)="onHidden()" bsModal #autoShownModal="bs-modal" class="modal fade" tabindex="-1" role="dialog" aria-hidden="true">
+<div *ngIf="isModalShown" [config]="{ show: true }" (onHidden)="onHidden()" bsModal #autoShownModal="bs-modal" class="modal fade"
+     tabindex="-1" role="dialog" aria-labelledby="dialog-auto-name" aria-modal="true">
```
Comment (IlyaSurmay) [inline demo/src/app/components/+modal/demos/auto-shown/auto-shown.html]: use renderer to append `aria-modal` attribute

**Review by IlyaSurmay — CHANGES_REQUESTED** [review]

**Review by IlyaSurmay — APPROVED** [review]

**Comment by YevheniiaMazur** [comment]
No modal has tag `aria-labelledby`, but example has.
Demo ngx bootstrap in this PR:
noarialabel

Example demo:
labelledby

**Comment by YevheniiaMazur** [comment]
Tested, looks good

**Review by valorkin — APPROVED** [review]

**Pull request merged**